### ISSUE TITLE
When should I (or not) use loss_weights ?

### ISSUE THREAD

**Issue by athtareq** [lululxvi/deepxde#908]
Hello, I've been seeing loss_weights used in quite a lot of codes and examples, and as far as I can deduce it just multiplies the loss with the specified number ? Obviously I'm wrong but can anyone give me further information ? Thanks !

**Comment by praksharma** [comment]
The loss weights are helpful is many cases. For example, NVIDIA Modulus uses a loss weight of 0.5 for the PDE loss for all problems.

> and as far as I can deduce it just multiplies the loss with the specified number ?

That is true.

**Comment by athtareq** [comment]
@praksharma thanks for the reply ! So when can I use it for example ?

**Comment by praksharma** [comment]
neural networks are black boxes. Nobody can give you definitive answers. Most probably you should use adaptive weights techniques for imbalanced loss terms.

**Comment by athtareq** [comment]
@praksharma duly noted, thank you very much.

**Comment by FZUcipher** [comment]
@praksharma Does DEEPXDE have any examples to implement adaptive weights？Thanks！

**Comment by praksharma** [comment]
I don't think so. I think adaptive weights is not very useful. It tries to minimise the training loss, thus minimising the weights perturbation (dJ/dw) in each iteration/epoch. I have seen PINNs converging early with  adaptive loss while the actual solution wasn't obtained. This mostly happens with forward problems with discontinuous boundary conditions and in ill-posed inverse problems.

**Comment by lululxvi** [comment]
@praksharma I agree with your comments. That is why we haven't implemented it in DeepXDE yet.
